src/backend/users/views.py

```
@@ -1,46 +1,60 @@
+        <p>BHA</p>
+        """.format(
+            login_link
+        )
+
+        # TODO replace with actual values parameterized based on environment
+        # issue 485 (https://github.com/azavea/echo-locator/issues/485)
+        send_mail(
+            "Your ECHO Login Link",
+            html_message,
+            "admin@domain.com",
+            [email],
+            fail_silently=False,
+            html_message=html_message,
+        )
+    except User.DoesNotExist:
+        raise Exception("User does not exist. Login email not sent.")
 
 
 class LoginPage(APIView):
     def post(self, request, **kwargs):
-        email = request.data["email"]
-        # try to find the User, but for privacy reasons, do not send 500 error back if not found
         try:
-            user = User.objects.get(username=email)
-            login_token = utils.get_query_string(user)
-            host = request.get_host()
-            protocol = "https://" if request.is_secure() else "http://"
-            login_link = protocol + host + reverse("obtain_token") + login_token
-
-            html_message = """
-            <p>Hi there,</p>
-            <p>Thanks for using ECHO! Here is your <a href="{}">link to login</a>. </p>
-            <p>BHA</p>
-            """.format(
-                login_link
-            )
-
-            # TODO replace with actual values parameterized based on environment
-            # issue 485 (https://github.com/azavea/echo-locator/issues/485)
-            send_mail(
-                "Your ECHO Login Link",
-                html_message,
-                "admin@domain.com",
-                [email],
-                fail_silently=False,
-                html_message=html_message,
-            )
-        except User.DoesNotExist:
+            # Will throw exception if cannot find the User
+            # For privacy reasons, do not send 500 error back if not found
+            send_login_link(request)
+        except Exception:
```

```python
except Exception:
    pass
```
Can be kind of a dangerous thing to have in a file because it'll catch and attempt to ignore pretty much any conceivable problem that could occur in the `try` block. This can make it harder to learn about and diagnose issues because there won't be any logging information available. For this reason I'd revert this to `except User.DoesNotExist` so that only that error is ignored and allow any other exceptions to bubble up so that it yells at us if there are other problems beyond what we specifically know we want to ignore.

---

Oh jeez, ok will definitely revert this back.